src/libraries/System.Private.CoreLib/src/System/String.cs

```diff
@@ -575,9 +575,9 @@ public StringRuneEnumerator EnumerateRunes()
             return new StringRuneEnumerator(this);
         }
 
-        internal static unsafe int wcslen(char* ptr) => SpanHelpers.IndexOfNullCharacter(ref *ptr);
+        internal static unsafe int wcslen(char* ptr) => SpanHelpers.IndexOfNullCharacter(ptr);
 
-        internal static unsafe int strlen(byte* ptr) => SpanHelpers.IndexOfNullByte(ref *ptr);
+        internal static unsafe int strlen(byte* ptr) => SpanHelpers.IndexOfNullByte(ptr);
```

Comment:
a nit: we can use `SpanHelpers.IndexOfNullByte` directly at call-sites and delete these wrappers to promote the usage of same API everywhere (CoreLib has total eight call-sites of `wcslen` and `strlen` combined).

Reply:
> a nit: we can use `SpanHelpers.IndexOfNullByte` directly at call-sites and delete these wrappers to promote the usage of same API everywhere (CoreLib has total 21 call-sites of `wcslen` and `strlen` combined).

It's a bit more complicated too delete it - e.g. VM relies on it for IL marshallers, mono seems too

Reply:
Ah corelib.h, forgot to check that. 😁
(they are used in `ilmarshalers.cpp` via `METHOD__STRING__STRLEN` and `METHOD__STRING__WCSLEN` macros)

Reply:
> Calls in C# code can use SpanHelpers directly.

I wonder if in this case ILLink will trim `strlen`/the other one as unused and then VM will crash, so probably better to leave as is or rename completely?

Reply:
For CoreCLR, ILLink keeps everything referenced by VM. I am not sure about Mono.